### PR TITLE
master: Use node label instead of node-role

### PR DESCRIPTION
**Pull request by MarcelMue**
With kubernetes 1.16 the `node-role` label is deprecated and we switch to the `node` label instead

towards https://github.com/giantswarm/giantswarm/issues/7428 and https://github.com/giantswarm/giantswarm/issues/7824

### CODE REVIEW

**Comment by tfussell** [comment]
Looks like `MasterNodeRole` can be deleted now.

File: pkg/label/k8s.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package label
 
 const (
+	// MasterNode label denotes a K8s cluster master node.
+	MasterNode = "node.kubernetes.io/master"
 	// MasterNodeRole label denotes K8s cluster master node role.
 	MasterNodeRole = "node-role.kubernetes.io/master"
```
Comment (omegas27) [inline pkg/label/k8s.go]: Any reason why you did not remove the MasterNodeRole here?

Comment (MarcelMue) [reply pkg/label/k8s.go]: I thought it was still in use somewhere but I guess it is not.

**Comment by kopiczko** [comment]
You may also cherry-pick to `legacy`. BTW for a while there is no way to test `master` changes. We should deal with that today.

**Comment by tfussell** [comment]
We have a systemd unit which applies the old label now. This no longer needs to be changed so I'm closing this PR.

**Pull request closed without merging**